### ISSUE TITLE
Exercise is completed (and counted correct) without giving the answer of the question.

### ISSUE THREAD

**Issue by swapnil106111** [learningequality/kolibri#3012]
### Observed behavior
When learner is trying to answer of such questions which are showing error while loading. After clicking on next question button, it is going to the next question without answering. For more refer a video which we have attached.

Channel ID : 5d132ba360b54b2ba9df169ffaa43ccb (test 2)

### Expected behavior
It should not increase points without answering the questions.

### Steps to reproduce
1. Import channel 5d132ba360b54b2ba9df169ffaa43ccb.
2. Login with learner.
3. Solve the exercise.
4. Path : test2 - Challenge Set: Multiplying Fractions with Fractions - 

### Context
Kolibri version: kolibri 0.7.0
Operating system: ubuntu 14.04
Browser: Chrome 63.0.3239.111

### Screenshots
![1](https://user-images.githubusercontent.com/12776071/34377400-dbc0565e-eb16-11e7-80e4-cf20f4ac1115.jpg)
![2](https://user-images.githubusercontent.com/12776071/34377401-dc212d94-eb16-11e7-82e1-641691322534.jpg)
![3](https://user-images.githubusercontent.com/12776071/34377403-dc91ab5a-eb16-11e7-81a0-c59fbedbd84e.jpg)

[VID_20171227_144408_s01.mp4.zip](https://github.com/learningequality/kolibri/files/1588759/VID_20171227_144408_s01.mp4.zip)

**Comment by jonboiser** [comment]
On the video,

* The Exercise fails to load, and shows the "There was a problem..." alert.
* The "Check" button immediately goes to "Next question" and the Exercise is **automatically counted as correct**.
* User presses "Next question", and the same thing happens (exercise is automatically counted as correct).

It might make sense to remove the buttons and checkmarks if the exercise fails to load.

**Comment by indirectlylit** [comment]
> It might make sense to remove the buttons and checkmarks if the exercise fails to load.

sounds right to me

**Comment by christianmemije** [comment]
> It might make sense to remove the buttons and checkmarks if the exercise fails to load.

But we need to allow learners to move on to the next question, in case it's just this exercise that won't render? We just should not mark it as correct. 

https://github.com/learningequality/kolibri/blob/5a175319bc685a662d798dbbf95e6994c3b5cb51/kolibri/plugins/learn/assets/src/views/assessment-wrapper/index.vue#L372

**Comment by jonboiser** [comment]
Yeah, that sounds correct.

**Comment by mrpau-eugene** [comment]
@indirectlylit would it be better to disable/remove the buttons and checkmarks when the exercise fails to load and display something like: `There was an error showing this item. Please refresh the page.` instead of just indicating that there was an error?

**Comment by mrpau-eugene** [comment]
Could someone explain to me why was it needed to record attempt log even though an error has occurred?

https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/src/views/assessment-wrapper/index.vue#L383-L399

I was wondering why was this one necessary

https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/src/views/assessment-wrapper/index.vue#L389-L398

**Comment by mrpau-eugene** [comment]
Update:
I found out where the answers are stored. I think it's an upstream problem.

The correctAnswer in the json file is `1.37142857143`.

Evaluating `1 13/35 is equal to 1.3714285714285714`. No wonder why it doesn't match.

**Comment by rtibbles** [comment]
It looks like you don't have a space between the 1 and the fraction. Answers are stored as decimals and then compared to a certain level of accuracy. I don't think there is anything wrong with the question.

The reason for generating the attempt log was to prevent learners from getting blocked by exercises with mostly broken questions. This should no longer be the case so we can go with the conservative approach described above.

**Comment by mrpau-eugene** [comment]
@rtibbles it does have a space. but I think the screenshot looks small making it look like it doesnt hae one. I was getting right answers for other questions too

> I don't think there is anything wrong with the question.

yeah, there wasn't anything wrong but I just noticed that the other items that have correct answers only have 2 decimal places e.g. `1.65` and `1.08` so I don't think there would be any problems comparing the answer of the learner to the correct answer.

> The reason for generating the attempt log was to prevent learners from getting blocked by exercises with mostly broken questions. This should no longer be the case

does this mean there aren't anymore bad questions?


> so we can go with the conservative approach described above.

which approach are you talking about? you mean marking the item as not complete or removing the buttons and check marks when an exercise fails to load?

**Comment by rtibbles** [comment]
Not marking the item as complete, but allowing the user to the next item.

**Comment by rtibbles** [comment]
*to progress

**Comment by mrpau-eugene** [comment]
@rtibbles I've been stuck for quite some time now.. what I did to reproduce this issue was:

1. Extract the `.perseus` file.
2. Removed a single quote from the "content" so it would fail as expected.
```json
{
    "question": {
        "content: "$\\ Multiply\\ and\\ express\\ in\\ mixed\\ fractions:$  \n$\\frac{4}{5}\\times\\ \\frac{12}{7}$\n\n[[☃ numeric-input 1]]",
        "images": {},
        ...
}
```
3. Compressed the file again

Now, after reproducing the issue, if it first loads the exercise that hasn't been modified, it would work fine. After hitting next, the modified exercise (the one I removed a single quote) would fail as expected. And after hitting next, the proceeding questions would also fail even though I did not modify them.

Was the kolibri exercise perseus plugin supposed to work this way?
